src/v/pandaproxy/schema_registry/handlers.cc

```diff
@@ -84,13 +75,10 @@ put_config(server::request_t rq, server::reply_t rp) {
       rq.req->content.data(), put_config_handler<>{});
     rq.req.reset();
 
-    auto res = rq.service().schema_store().set_compatibility(config.compat);
-    if (res.has_error()) {
-        rp.rep = make_errored_body(res.error());
-        co_return rp;
-    }
+    auto res
+      = rq.service().schema_store().set_compatibility(config.compat).value();
```

Comment:
> I take it old semantics of return error body are now handled by exception being thrown when trying to extract the value?

This is correct. If the `result` contains an error, it will throw a `schema_registry::exception` constructed from the `schema_registry::error_info`which contains the `error_code` and a message. This allows the final `exception_reply` to contain an error message with nice context (such as which subject or schema id failed).